### PR TITLE
Generate Template link and blob container dropdown fix

### PR DESCRIPTION
**Pull request by sahuroshan**
As part of this PR 3 bugs have been fixed.
1. generate template link bugs.
In the 3 page as well now latest sku recommendation are fetched(earlier the recommended sku was displayed based on older sku recommendation) and a **loading... text is displayed until the latest sku recommendation are fetched** . Doing this change fixed both of the bugs related to generate template link.

image

2. blob container dropdown bug.
There was another bug , the ARM template was getting uploaded to the first blob container in the dropdown. Added an ondropdownchange event to fix the same.

ADO bug links-
https://msdata.visualstudio.com/Database%20Systems/_workitems/edit/3044065
https://msdata.visualstudio.com/Database%20Systems/_workitems/edit/3044076

### CODE REVIEW

File: extensions/sql-migration/src/dialog/skuRecommendationResults/SelectStorageAccountDialog.ts [hunk 1]
```diff
@@ -365,6 +361,16 @@ export class SelectStorageAccountDialog {
 				CSSStyles: { ...DROPDOWN_CSS },
 			}).component();
 
+		this._disposables.push(
+			this._blobContainerDropdown.onValueChanged(async (value) => {
+				if (value && value !== 'undefined') {
```
Comment (Ramudaykumar) [inline extensions/sql-migration/src/dialog/skuRecommendationResults/SelectStorageAccountDialog.ts:366]: Don't put undefined in quotes, this way it is compering with the literal 'undefined' not value undefined. it should be

value !== undefined

**Review by Ramudaykumar — CHANGES_REQUESTED** [review]
Looks good overall, I have added a comment, please take the change.

File: extensions/sql-migration/src/dialog/skuRecommendationResults/SelectStorageAccountDialog.ts (hunk 1 above)
Comment (sahuroshan) [reply extensions/sql-migration/src/dialog/skuRecommendationResults/SelectStorageAccountDialog.ts]: done

**Comment by sahuroshan** [comment]
> Looks good overall, I have added a comment, please take the change.

done

**Review by Ramudaykumar — APPROVED** [review]
Looks good to me.

**Pull request merged**